test/src/CBot/CBot_test.cpp

```diff
@@ -219,7 +252,7 @@ class CBotUT : public testing::Test
     }
 
 protected:
-    std::unique_ptr<CBotProgram> ExecuteTest(const std::string& code, CBotError expectedError = CBotNoErr)
+    std::unique_ptr<CBotProgram> ExecuteTest(const std::string& code, CBotError expectedError = CBotNoErr, bool stepMode=true)
```

Comment:
I made this change because try is "special": `try { ... } catch(condition) { ... }` only checks the condition when **not** in step mode. In step mode it simply skips over the condition. I guess this may have been done in an attempt to make debugger user experience a little more pleasant - you don't jump to catch and back after every step. I agree that in a perfect world the behaviour of the language would be the same in step mode and in regular mode.

https://github.com/colobot/colobot/blob/f07b7a1b60cd5685d7cd5f5677669423281fefa0/CBot/src/CBot/CBotInstr/CBotTry.cpp#L104-L105

Reply:
Demo: on master (and probably on dev) this program runs forever in normal mode, but halts in step mode:
```
extern void New()
{
	try {
		try {
			wait(0);
		} catch(Foo()) {
		}
	} catch(10) {
		while(true);
	}
}

bool Foo() {
	throw 10;
	return false;
}
```

Maybe this hack should be removed from try ... catch. Then my tests will work in step mode. What do you think?

Reply:
In that case we should probably test and fix ```try...catch...finally``` first because it basically does what needs to already.  How to stop long running external calls is a separate issue and needs very few if any changes in the cbot side of things. Maybe @immibis doesn't mind if we cherrypick their [try...catch...finally unit tests.](https://github.com/immibis/colobot/commit/b36075f7e131d9dd20261a86681e874ab4c7ef85)

Reply:
I see **no reason** why `try ... catch(ErrorCode) ... finaly` where `ErrorCode` is a constant would not work just fine in step mode. The hack specifically affects `try ... catch(expression)` in cases where we can detect the evaluation of the expression (such as when it is an infinite loop as in my "Demo" example above). I'll cherry-pick the tests from https://github.com/immibis/colobot/tree/cbot-cleanup into a separate pull request. Stay tuned.